Résolution du bug de la barre de progression

BaboucheOne opened the pull request:
Résolution du bug de la barre de progression

- Il est désormais possible de drag la bar de progression sans souci.
- Simplification de la formule pour trouver le pourcentage a l'horizontal.

BaboucheOne committed:
Fix de la progress bar

Il est desormais possible de drag la bar de progression sans soucie, ainsi que de cliquer n importe pour avancer jusqu'au curseur

MaitreMenard commented on BaseStation/src/openrocket_simulation.py:
```diff
@@ -15,4 +15,4 @@ def __init__(self, filename):
                 self.altitude.append(float(data[1]))
         except ValueError:
-            raise DomainError("Le fichier de simulation " + filename + " n'a pas pu être chargé")
+            raise DomainError("Le fichier de simulation " + filename + " n'a pas pu être chargé")
```
Le PEP8 recommande que les fichiers soient terminés par des sauts de ligne, car c'est une convention sur les systèmes Unix.

MaitreMenard commented on BaseStation/src/ui/ExtendedQSlider.py:
```diff
@@ -21,12 +21,13 @@ def mousePressEvent(self, q_mouse_event):
             q_mouse_event.accept()
 
-        super().mousePressEvent(q_mouse_event)
+        super().mousePressEvent(q_mouse_event)
```
Même chose que le premier commentaire.

MaitreMenard commented on BaseStation/src/ui/ExtendedQSlider.py:
```diff
@@ -21,12 +21,13 @@ def mousePressEvent(self, q_mouse_event):
                 new_value = self.minimum() + (value_range * (self.height() - q_mouse_event.y())) / self.height()
             else:
-                new_value = self.minimum() + (value_range * q_mouse_event.x()) / self.width()
+                new_value = q_mouse_event.x() / self.width() * 100
```
Ceci ne fonctionne que si le range est entre 0 et 100. Ça ne fonctionnera plus quand le range va être modifié en fonction du nombre de packets.

MaitreMenard commented on BaseStation/src/ui/ExtendedQSlider.py:
```diff
@@ -21,12 +21,13 @@ def mousePressEvent(self, q_mouse_event):
+                new_value = q_mouse_event.x() / self.width() * 100
 
             if self.invertedAppearance():
                 self.setValue(self.maximum() - new_value)
```
L'arrondissement devrait aussi être appliqué ici.

BaboucheOne replied:
Effectivement je n avais pas pris en compte le nombre de packets sur le moment.

MaitreMenard commented:
J'ai testé le fonctionnement de la barre et le curseur ne reste plus bloqué quand on glisse la souris au dessus. Par contre, le curseur bloque encore parfois quand on le drag. J'ai l'impression qu'on va devoir reimplémenter la méthode _mouseMoveEvent_ (voir: http://doc.qt.io/qt-5/qwidget.html#mouseMoveEvent).

BaboucheOne commented:
Comment test-tu la barre ? 
Apres plusieurs essai, la barre répond comme demandé.

BaboucheOne committed:
Fix des bug sur la ExtendedQSlider

Reimplantation de la methode 'mouseMoveEvent'.

Reecriture de la methode 'mousePressEvent'.

Creation d'une methode 'compute_position' qui evite la duplication de code ains de calculer la position de la barre en fonction du curseur.

MaitreMenard commented:
Dans le fichier _openrocket\_simulation.py_, tu as mis le saut de ligne à la fin de la ligne 17 et non à la fin de la ligne 18. À part ça tout est beau.

BaboucheOne committed:
Update openrocket_simulation

Mise du saut de ligne
